krun/scheduler.py

```diff
@@ -228,17 +230,22 @@ def run(self, mailer, dry_run=False):
                 flag = "C"
             except util.ExecutionFailed as e:
                 util.log_and_mail(mailer, error, "Benchmark failure: %s" % self.key, e.message)
-                measurements = EMPTY_MEASUREMENTS
+                measurements = self.empty_measurements
                 flag = "E"
 
             if vm_def.instrument:
-                instr_data = vm_def.get_instr_data()
-                for k, v in instr_data.iteritems():
-                    assert len(instr_data[k]) == in_proc_iters
+                if flag != "E":
+                    instr_data = vm_def.get_instr_data()
+                    for k, v in instr_data.iteritems():
+                        assert len(instr_data[k]) == in_proc_iters
+                else:
+                    # Although there may be some instrumentation data, it is
```

Comment:
I think this comment might be clearer as "As an error happened, we have to assume that any instrumentation data is compromised; we thus ignore it."